runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/FlinkStreamingTranslationContext.java

```diff
@@ -99,12 +102,21 @@ public void setCurrentTransform(AppliedPTransform<?, ?, ?> currentTransform) {
 
 
   @SuppressWarnings("unchecked")
-  public <T extends PInput> T getInput(PTransform<T, ?> transform) {
-    return (T) currentTransform.getInput();
+  public <T extends PValue> T getOnlyInput(PTransform<T, ?> transform) {
+    return (T) Iterables.getOnlyElement(currentTransform.getInputs()).getValue();
+  }
+
+  public <T extends PInput> List<TaggedPValue> getInput(PTransform<T, ?> transform) {
```

Comment:
Should this be `getInputs()`?

Same holds for `getOutput()` and the corresponding methods in `FlinkBatchTranslationContext`.